File: Packages/com.battlement.client/Runtime/Protocol/RenderOrder.cs
#nullable enable

namespace Battlement
{
    /// <summary>Whether ordering applies to a visual group or a single renderer.</summary>
    public enum RenderOrderKind : byte
    {
        Group,
        Layer,
    }

    /// <summary>Visual order relative to the nearest enclosing sorting group.</summary>
    public readonly struct RenderOrder
    {
        public RenderOrder(RenderOrderKind kind, short order) => (Kind, Order) = (kind, order);

        public RenderOrderKind Kind { get; }
        public short Order { get; }
    }
}

File: crates/battlement/src/render_order.rs
//! Relative visual ordering for world renderers and sorting groups.

/// Ordering inside the nearest enclosing sorting group, or the scene when ungrouped.
#[derive(Clone, Copy, Debug, Eq, PartialEq)]
pub enum RenderOrder {
  /// Render this object's visual descendants together at the declared relative order.
  Group(i16),
  /// Place this object's renderer within its enclosing sorting group.
  Layer(i16),
}
